### PR TITLE
Integration tests

### PR DESCRIPTION
**Pull request by emmiiorji**
This change adds test for the index and show views of posts and users.

### CODE REVIEW

File: spec/features/posts/show.html.erb_spec.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    post_text = find('.post-text')
+    expect(post_text).to have_content(@post.text)
+  end
+
+  it 'shows comments of each commentors' do
+    comments = @post.comments
+    comments.each do |comment|
+      expect(page).to have_content(comment.text)
+    end
+  end
```
Comment (gemmen29) [inline spec/features/posts/show.html.erb_spec.rb:43]: - [ ] Kindly also test if I can see the username of the commentator as per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md#:~:text=I%20can%20see%20the%20username%20of%20each%20commentor.).

File: spec/features/users/show.html.erb_spec.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    expect(page).to have_content("Number of posts: #{@user.posts_counter}")
+    expect(page).to have_content(@user.bio)
+  end
+
+  it 'shows the last three posts' do
+    @posts.each do |post|
+      expect(page).to have_content(post.title)
+    end
+    expect(page).to_not have_content(@post1.title)
+  end
```
Comment (gemmen29) [inline spec/features/users/show.html.erb_spec.rb:32]: - [ ] Kindly provide tests that represent those [requirements](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md#:~:text=I%20can%20see%20a,user%27s%20post%27s%20index%20page.).

File: app/controllers/posts_controller.rb [hunk 3]
```diff
@@ -2,7 +2,7 @@ class PostsController < ApplicationController
   def index
     @users = User.all
     @user = User.find_by(id: params[:user_id])
-    @posts = @user ? Post.where(author: @user) : nil
+    @posts = @user ? Post.where(author: @user).paginate(page: params[:page], per_page: 5) : nil
```
Comment (gemmen29) [inline app/controllers/posts_controller.rb:5]: - [ ] ~~[DEPRECATED] Kindly solve the `N+1` problem when fetching the posts and their comments as per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md#:~:text=Make%20sure%20that%20the%20N%2B1%20problem%20is%20solved%20when%20fetching%20all%20posts%20and%20their%20comments%20for%20a%20user.%20This%20is%20the%20corresponding%20wireframe).~~

**Review by gemmen29 — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 

## Required Changes ♻️
_Check the comments under the review._



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 3 above)
Comment (emmiiorji) [reply app/controllers/posts_controller.rb:5]: Hi @gemmen29,

Thanks for the kind review.

We don't seem to get a warning from bullet about it.

Comment (gemmen29) [reply app/controllers/posts_controller.rb:5]: Hi @emmiiorji 

I don't get what you are trying to say.

Could you please try to explain more?

Comment (emmiiorji) [reply app/controllers/posts_controller.rb:5]: Okay @gemmen29.

We have the [bullet gem](https://github.com/flyerhzm/bullet) installed which checks for n+1 queries. We've solved the issues it identified as n+1 queries.

Comment (gemmen29) [reply app/controllers/posts_controller.rb:5]: Hi @emmiiorji,

I think you are right the `bullet gem` solve the `N + 1` problem.
Ignore this comment and work on other suggestions and I will cross out this comment.

**Review by Davidosky007 — APPROVED** [review]
Hi @emmiiorji 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
* All tests passed 👍 
* No linter errors 👍 
* Good looking readme file 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 3 above)
Comment (emmiiorji) [reply app/controllers/posts_controller.rb:5]: Okay, thanks.

**Pull request merged**